src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c

```diff
@@ -35,54 +39,473 @@ const ActorInit En_Stone_heishi_InitVars = {
+        func_80151BB4(play, 0x35);
+        func_80151BB4(play, 0x12);
+        this->action = EN_STONE_ACTION_2;
+        this->actionFunc = func_80BC9680;
+    } else {
+        func_800B8500(&this->actor, play, 400.0f, 400.0f, PLAYER_AP_MINUS1);
+    }
+}
+
+void EnStoneheishi_Update(Actor* thisx, PlayState* play) {
+    EnStoneheishi* this = THIS;
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->actionFunc(this, play);
+
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 20.0f, 20.0f, 50.0f, 29);
+    Actor_SetScale(&this->actor, 0.01f);
+
+    if (((gSaveContext.save.weekEventReg[0x29] & 0x40) || (play->actorCtx.unk4 == 100)) &&
+        (((s32)player->stateFlags1 << 8) >= 0)) {
+        if ((this->animIndex != EN_STONE_HEISHI_WAVE) &&
+            ((((this->action == EN_STONE_ACTION_1) || (this->action == EN_STONE_ACTION_2)) ||
+              (this->action == EN_STONE_ACTION_CHECK_ITEM)) ||
+             ((this->action == EN_STONE_ACTION_DRINK_BOTTLE) &&
+              (this->DrinkBottleState <= EN_STONE_DRINK_BOTTLE_DRINKING)))) {
+            EnStoneheishi_HeadTowardsLink(this);
+        } else {
+            this->targetHeadPosX = 0;
+        }
+
+        if (!(gSaveContext.save.weekEventReg[0x29] & 0x40)) {
+            Actor_SetFocus(&this->actor, 30.0f);
+        } else {
+            Actor_SetFocus(&this->actor, 60.0f);
+        }
 
-extern UNK_TYPE D_06000768;
-extern UNK_TYPE D_06003BFC;
+        Math_SmoothStepToS(&this->headRotX, this->targetHeadPosX, 1, 0xBB8, 0);
+        Math_SmoothStepToS(&this->headRotY, this->targetHeadPosY, 1, 0x3E8, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stone_heishi/EnStoneheishi_Init.s")
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stone_heishi/EnStoneheishi_Destroy.s")
+s32 EnStoneheishi_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                                   Gfx** gfx) {
+    EnStoneheishi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stone_heishi/func_80BC935C.s")
+    if (limbIndex == SOLDIER_LIMB_HEAD) {
+        rot->x += this->headRotX;
+        rot->y += this->headRotY;
+        rot->z += this->headRotZ;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stone_heishi/func_80BC941C.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stone_heishi/func_80BC94B0.s")
+void EnStoneheishi_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
+    static Vec3f D_80BCA404 = { 0.0f, 0.0f, 0.0f };
+    EnStoneheishi* this = THIS;
+    Gfx* gfx2;
```

Comment:
```suggestion
  void EnStoneheishi_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfxp) {
    static Vec3f D_80BCA404 = { 0.0f, 0.0f, 0.0f };
    EnStoneheishi* this = THIS;
    Gfx* gfx;
```
This is the most comment convention